### PR TITLE
feat(dashboard.volumes): add acl tab

### PR DESCRIPTION
**Pull request by antleblanc**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/netapp`
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-7152
| License          | BSD 3-Clause

### :sparkles: Features

47639c4 - feat(dashboard.volumes): add acl tab

Signed-off-by: Antoine Leblanc <antoine.leblanc@ovhcloud.com>

### CODE REVIEW

File: packages/manager/modules/netapp/package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "@ovh-ux/ng-at-internet-ui-router-plugin": "^3.1.2",
     "@ovh-ux/ng-ovh-utils": "^14.0.14",
     "@ovh-ux/ng-translate-async-loader": "^2.1.0",
+    "@ovh-ux/ng-ui-router-breadcrumb": "^1.1.4",
```
Comment (marie-j) [inline packages/manager/modules/netapp/package.json:31]: Maybe add to generator ?

File: packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/controller.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  addAcl() {
+    this.isAdding = true;
+    this.currentAcl = {
+      accessType: null,
+      accessTo: null,
+      accessLevel: null,
+    };
+    this.acls.push({});
+    this.selectRow(this.acls.length - 1);
```
Comment (marie-j) [inline packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/controller.js:35]: Let's say I have 4 rows, So previous selected row would be row number 4. Here we add an element so we now have 5 rows but we still select row number 4 ?

File: packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/routing.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('netapp.dashboard.volumes.dashboard.acl', {
+    url: '/acl',
+    component: 'ovhManagerNetAppVolumesDashboardAcl',
+    resolve: {
+      acls: /* @ngInject */ ($http, serviceName, volumeId) =>
+        $http
+          .get(`/storage/netapp/${serviceName}/share/${volumeId}/acl`)
+          .then(({ data }) => data)
+          .catch(() => []),
```
Comment (marie-j) [inline packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/routing.js:10]: Should we stil display view if acls returns an error, in that case I'm not sure about this behaviour

File: packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/routing.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      breadcrumb: /* @ngInject */ () => 'acl',
```
Comment (marie-j) [inline packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/routing.js:11]: Not translated ?

File: packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/template.html [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    <oui-button
+        variant="secondary"
+        disabled="$ctrl.isAdding"
+        on-click="$ctrl.addAcl()"
+        icon-left="oui-icon-add"
+        class="mb-3"
+    >
+        <span data-translate="netapp_volumes_acl_add"></span>
+    </oui-button>
+    <oui-datagrid rows="$ctrl.acls" ng-if="$ctrl.acls.length">
```
Comment (marie-j) [inline packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/template.html:20]:
```suggestion
    <oui-datagrid rows="$ctrl.acls" >
```
You could still display the datagrid but empty no ?

Comment (antleblanc) [reply packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/template.html:20]: Assuming that there's no `aclRule` already configured, the expected behavior is
that we have to hide the datagrid. Once the user click on "Add a new ACL", then
we have to display the datagrid.

File: packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/routing.js (hunk 3 above)
Comment (antleblanc) [reply packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/routing.js:10]: You want me to redirect instead to the volume dashboard with an error message?

File: packages/manager/modules/netapp/package.json (hunk 1 above)
Comment (antleblanc) [reply packages/manager/modules/netapp/package.json:31]: Good point! I will handle it separately (new Pull Request).

Comment (antleblanc) [reply packages/manager/modules/netapp/package.json:31]: Please refer to #5430

File: packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/routing.js (hunk 3 above)
Comment (marie-j) [reply packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/routing.js:10]: Or let the error page handle it

**Review by marie-j — APPROVED** [review]

File: packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/controller.js (hunk 2 above)
Comment (antleblanc) [reply packages/manager/modules/netapp/src/dashboard/volumes/dashboard/acl/controller.js:35]: Seen together.

**Pull request merged**